nexus/src/app/background/sync_switch_configuration.rs

```diff
@@ -1265,6 +1341,17 @@ async fn apply_switch_port_changes(
                     }
                 };
 
+                if config_on_switch.into_inner() == dpd_port_settings {
```

Comment:
I think this will have the desired effect. However overall implementation here is not as clear as it could be. What we want is to calculate two sets: `O` are the switch port settings in the omicron database, `D` are the switch port settings in dendrite. We can do the following operations hashing just on the switch/port names. This gives us the sets of port settings to add and remove. `/` denotes set difference and `&` denotes set intersection in what follows.

- `to_add = O / D`
- `to_del = D / O`

Additionally we have settings that may have been modified that need to be checked.

- `to_check = D & O`

We reduce these to the set that actually needs modification with a filter function

```
to_modify = to_check.filter(|d, o| d != 0)
```

I'm noting this as the filtering in this function (pretty late in the overall process) feels out of place. Looking at the `SwitchPortSettingsManager::Changes` function sheds light on why. That function is not actually computing changes, or a difference in the desired state versus the target state, it's just computing the desired state and then we get here and have to filter through things as we're making the changes.

Like I said above, I think this will have the desired effect, but I think we should consider some follow-on refactoring to simplify things a bit.

Reply:
I think that's a fair assessment. The initial implementation was focused on ensuring we're always enforcing what is in Nexus / Omicron, and then additional passes were made to make things more efficient (with this case unfortunately getting overlooked).